open-api/rest-catalog-open-api.yaml

```diff
@@ -3103,6 +3141,32 @@ components:
         uuid:
           type: string
 
+    Credential:
+      type: object
+      required:
+        - prefix
+        - config
+      properties:
+        prefix:
+          type: string
+          description: Indicates a storage location prefix where the credential is relevant. Clients should choose the most
+            specific prefix if several credentials of the same type are available.
+        config:
```

Comment:
```suggestion
        expiresAt:
          type: integer
          format: int64
          description: Timestamp in milliseconds since epoch (1970-01-01-00:00:00Z) when the credentials expire. If credentials do not expire, this field is absent.
        refreshNotBefore:
          type: integer
          format: int64
          description: If this field is present, clients must not refresh credentials before this timestamp in milliseconds since epoch (1970-01-01-00:00:00Z). This field must not be present for credentials that do not expire.
        config:
```

Reply:
Those things should not be part of the OpenAPI spec and be rather specific to the underlying storage provider and should be documented along with other properties related to that storage provider.

Reply:
> rather specific to the underlying storage provider

The whole effort is about _refreshing_ expiring credentials, so such an attribute is a first-class citizen, not something hidden in a rather generic property bag.

Reply:
It makes sense to me to have an "expires at" field of some sort. It might help make the refresh implementation logic more reusable across different providers?

Reply:
we discussed this in yesterday's catalog sync and concluded that we don't want to introduce a separate `expires-at-ms` field to the spec.